zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/IdentitySetupInitializeProcessor.java

```diff
@@ -0,0 +1,124 @@
+import io.camunda.zeebe.protocol.impl.record.value.authorization.AuthorizationRecord;
+import io.camunda.zeebe.protocol.impl.record.value.authorization.IdentitySetupRecord;
+import io.camunda.zeebe.protocol.impl.record.value.authorization.Permission;
+import io.camunda.zeebe.protocol.impl.record.value.authorization.RoleRecord;
+import io.camunda.zeebe.protocol.record.intent.AuthorizationIntent;
+import io.camunda.zeebe.protocol.record.intent.IdentitySetupIntent;
+import io.camunda.zeebe.protocol.record.intent.RoleIntent;
+import io.camunda.zeebe.protocol.record.intent.UserIntent;
+import io.camunda.zeebe.protocol.record.value.AuthorizationOwnerType;
+import io.camunda.zeebe.protocol.record.value.AuthorizationResourceType;
+import io.camunda.zeebe.protocol.record.value.EntityType;
+import io.camunda.zeebe.protocol.record.value.PermissionType;
+import io.camunda.zeebe.protocol.record.value.RoleRecordValue;
+import io.camunda.zeebe.protocol.record.value.UserRecordValue;
+import io.camunda.zeebe.stream.api.records.TypedRecord;
+import io.camunda.zeebe.stream.api.state.KeyGenerator;
+
+@ExcludeAuthorizationCheck
+public final class IdentitySetupInitializeProcessor
+    implements DistributedTypedRecordProcessor<IdentitySetupRecord> {
+  private final RoleState roleState;
+  private final UserState userState;
+  private final StateWriter stateWriter;
+  private final KeyGenerator keyGenerator;
+  private final CommandDistributionBehavior commandDistributionBehavior;
+
+  public IdentitySetupInitializeProcessor(
+      final ProcessingState processingState,
+      final Writers writers,
+      final KeyGenerator keyGenerator,
+      final CommandDistributionBehavior commandDistributionBehavior) {
+    roleState = processingState.getRoleState();
+    userState = processingState.getUserState();
+    stateWriter = writers.state();
+    this.keyGenerator = keyGenerator;
+    this.commandDistributionBehavior = commandDistributionBehavior;
+  }
+
+  @Override
+  public void processNewCommand(final TypedRecord<IdentitySetupRecord> command) {
+    final var key = keyGenerator.nextKey();
+    initializeDefaultEntities(key, command);
+    commandDistributionBehavior
+        .withKey(key)
+        .inQueue(DistributionQueue.IDENTITY)
+        .distribute(command);
+  }
+
+  @Override
+  public void processDistributedCommand(final TypedRecord<IdentitySetupRecord> command) {
+    initializeDefaultEntities(command.getKey(), command);
+    commandDistributionBehavior.acknowledgeCommand(command);
+  }
+
+  private void initializeDefaultEntities(
+      final long commandKey, final TypedRecord<IdentitySetupRecord> command) {
+    final var roleRecord = command.getValue().getDefaultRole();
+    final var userRecord = command.getValue().getDefaultUser();
+
+    final var roleExists = roleState.getRoleKeyByName(roleRecord.getName()).isPresent();
+    final var userExists = userState.getUser(userRecord.getUsername()).isPresent();
```

Comment:
There is only this 1 processor 😉 It is writing events not followup commands. I had to do that because if I use a followup command I cannot access the keys of the role/user here so I wouldn't know the data I need to assign the two.

Reply:
Right, it actually make sense 😄